### PR TITLE
Consistently start Point rows/columns at 0, not 1

### PR DESCRIPTION
**Pull request by StephenChan**
For issue #314.

- Added a data migration to convert existing points and annotation areas to 0-indexing.

    - The annotation areas defined by pixel counts, not percentages, need to be converted.
    
    - As of a few days ago, there were no out-of-range point locations in production (assuming 1-indexing). I'll double-check again before running this migration in production though.
    
    - The migration runs in 4 minutes in staging. Staging has 10m annotations, and production has 50m annotations, so I'd budget maybe 30 minutes for running the migration in production.

- Updated the following to use 0-indexing:

    - Point generation.
    - Individual-image annotation area editing.
    - `removepointoutliers` management command. Also sped up the command with F expressions.
    - Bounds checks in annotation upload by CSV.
    - Annotation upload by CPC.
    - Annotation export to CPC.
    
- Updated patch generation and annotation tool code as discussed in issue #314. To recap, patch generation was off by half a pixel, while annotation tool code had a mix of indexing schemes and another half-pixel issue.

- The API was already on 0-indexing, even the docs. So no changes there. (Yes, this means I was confused on the indexing as recently as a couple months ago.)

- Added row/column bounds checks to the `Point` model's `save()` method.

Let me know if you can think of any other row/column code to double check.

### CODE REVIEW

File: project/images/models.py [hunk 1]
```diff
@@ -780,6 +790,20 @@ def machine_confidence(self):
             # No scores means max(scores) will raise this
             return 0
 
+    def save(self, *args, **kwargs):
```
Comment (beijbom) [inline project/images/models.py:793]: nice to see this here.

File: project/visualization/utils.py [hunk 2]
```diff
@@ -111,33 +111,42 @@ def generate_patch_if_doesnt_exist(point_id):
...
+    approx_region_size = int(max(image.original_width, image.original_height)
+                             * settings.LABELPATCH_SIZE_FRACTION)
 
-    # Load the image convert to RGB.
+    # Load the image and convert to RGB.
     im = PILImage.open(original_image_file)
     im = im.convert('RGB')
 
     # Crop.
+    # - Both CoralNet coordinates and Pillow coordinates start from 0 at the
```
Comment (beijbom) [inline project/visualization/utils.py:130]: this part looks good also.

**Comment by beijbom** [comment]
I only reviewed some of it, but the stuff I was thinking about are implemented. That check on the model makes me happy.

**Comment by StephenChan** [comment]
Thanks for looking at this @beijbom . I'll merge into master and rebase the `beta2-rollout` branch. I haven't written the blog post for these changes yet, but I'll certainly do that before updating production.

**Pull request merged**